nvbench/device_info.cu

```diff
@@ -18,14 +18,19 @@
 
 #include <nvbench/device_info.cuh>
 
+#include <nvbench/config.cuh>
 #include <nvbench/cuda_call.cuh>
 #include <nvbench/detail/device_scope.cuh>
+#include <nvbench/internal/nvml.cuh>
```

Comment:
The intention is:

- `detail`: Implementation details that need to be installed because they're included from public headers
- `internal`: Implementation details that are only included in cxx/cu files and don't need to be installed.

I agree that this separation isn't clear or well enforced at the moment -- cleaning this up in on my TODO list, but low priority.